mkr-mcd.md

```diff
@@ -181,62 +226,152 @@ This allows us to enforce properties after each step, and restore the old state
          <can> ... MSGSENDER |-> (CANADDRS => CANADDRS -Set SetItem(ADDRTO)) ... </can>
 ```
 
-### Ilk Initialization
+-   `Vat.consent` checks whether a transaction was beneficial for a given account, otherwise makes sure that `Vat.wish` is set.
+    This encodes that "rational actors consent to actions which benefit them".
+
+-   `Vat.safe` checks that a given `Urn` of a certain `ilk` is not over-leveraged.
+
+-   `Vat.nondusty` checks that a given `Urn` has either exactly 0 on a non-dusty amount of debt.
+    **TODO**: Currently we use `urnDebt ==Int 0`, whereas Solidity implementation uses `urnArt ==Int 0`.
+              Does it matter? They are equivalent as long as `urnRate =/=Int 0`.
+
+```k
+    syntax VatStep ::= "consent" Int Address
+ // ----------------------------------------
+    rule <k> Vat . consent _     ADDR => Vat . wish ADDR ... </k> [owise]
+    rule <k> Vat . consent ILKID ADDR => .               ... </k>
+         <vatStack>
+           ListItem ( <vat>-fragment
+                        _ _
+                        <ilks> ...   ILKID          |-> ILK' ... </ilks>
+                        <urns> ... { ILKID , ADDR } |-> URN' ... </urns>
+                        <gem>  ... { ILKID , ADDR } |-> COL' ... </gem>
+                        <dai>  ...           ADDR   |-> DAI' ... </dai>
+                        _ _ _ _ _
+                      </vat>-fragment
+                    )
+           ...
+         </vatStack>
+         <vat>
+           <ilks> ...   ILKID          |-> ILK ... </ilks>
+           <urns> ... { ILKID , ADDR } |-> URN ... </urns>
+           <gem>  ... { ILKID , ADDR } |-> COL ... </gem>
+           <dai>  ...           ADDR   |-> DAI ... </dai>
+           ...
+         </vat>
+      requires COL                  <=Int COL'
+       andBool DAI                  <=Int DAI'
+       andBool urnBalance(ILK, URN) <=Int urnBalance(ILK', URN')
+
+    syntax VatStep ::= "less-risky" Int Address
+ // -------------------------------------------
+    rule <k> Vat . less-risky ILKID ADDR => Vat . safe ILKID ADDR ... </k> [owise]
+    rule <k> Vat . less-risky ILKID ADDR => .                     ... </k>
+         <vatStack>
+           ListItem ( <vat>-fragment
+                        _ _
+                        <ilks> ...   ILKID          |-> ILK' ... </ilks>
+                        <urns> ... { ILKID , ADDR } |-> URN' ... </urns>
+                        _ _ _ _ _ _ _
+                      </vat>-fragment
+                    )
+           ...
+         </vatStack>
+         <vat>
+           <ilks> ...   ILKID          |-> ILK ... </ilks>
+           <urns> ... { ILKID , ADDR } |-> URN ... </urns>
+           ...
+         </vat>
+      requires urnBalance(ILK, URN) <=Int urnBalance(ILK', URN')
```

Comment:
"less risky" - anyone can interact with a CDP if it's to pay debt or add collateral.